std/string.d

```diff
@@ -3215,6 +3186,202 @@ S center(S)(S s, size_t width, dchar fillChar = ' ') @trusted pure
     });
 }
 
+/++
+    Center justify $(D r) in a field $(D width) characters wide. $(D fillChar)
+    is the character that will be used to fill up the space in the field that
+    $(D r) doesn't fill.
+
+    Params:
+        r = string or forward range of characters
+        width = minimum field width
+        fillChar = used to pad end up to $(D width) characters
+
+    Returns:
+        a lazy range of the center justified result
+
+    See_Also:
+        $(LREF leftJustifier)
+        $(LREF rightJustifier)
+  +/
+
+auto centerJustifier(Range)(Range r, size_t width, dchar fillChar = ' ')
+    if (isForwardRange!Range && isSomeChar!(ElementEncodingType!Range))
+{
+    alias C = Unqual!(ElementEncodingType!Range);
+
+    static if (C.sizeof == 1)
+    {
+        import std.utf : byDchar, byChar;
+        return centerJustifier(r.byDchar, width, fillChar).byChar;
+    }
+    else static if (C.sizeof == 2)
+    {
+        import std.utf : byDchar, byWchar;
+        return centerJustifier(r.byDchar, width, fillChar).byWchar;
+    }
+    else static if (C.sizeof == 4 && hasLength!Range)
+    {
+        import std.range : chain, repeat;
+
+        auto len = r.length;
+        if (len > width)
+            len = width;
```

Comment:
You can use `walkLength(r, up_to=width)` for ranges without length, then the explicit struct version below can go.